README.md

```diff
@@ -1,16 +1,110 @@
 
-A Clojure library designed to provide a collection of helper functions to support Clojure(script) graph parsers using
-om.next graph syntax.
+A Clojure library designed to help you write Clojure(script) graph query processing
+parsers for the query notation used by [Om.next](https://github.com/omcljs/om)
+and [Fulcro](https://github.com/fulcrologic/fulcro).  The query notation is closely related
+to [Datomic Pull syntax](https://docs.datomic.com/on-prem/pull.html).
 
-For an introduction on `om.next` parser development [please check my article on the subject](https://medium.com/@wilkerlucio/implementing-custom-om-next-parsers-f20ca6db1664).
-This library encapsulates the ideas presented there, and go deeper on parser patterns. And all documentation here assumes you understand the Om.next query syntax.
+For an introduction to general parser development
+[please check my article on the subject](https://medium.com/@wilkerlucio/implementing-custom-om-next-parsers-f20ca6db1664).
+This library encapsulates the ideas presented there.
+And all documentation here assumes you understand the Om.next query syntax.
 
-## Ideas presentation
+## Usage
 
-At Clojure days I presented the main ideas behind this library, you can check it at: https://www.youtube.com/watch?v=r3zywlNflJI.
+The library has a number of different usage models. One of the most commonly
+used features is Connect.  The basic idea is as follows:
 
-## Manual
+. You define *resolvers* that can take simple inputs (e.g. entity ID) and turn
+that into a set of outputs.
+. Your resolvers implies "edges" by outputting IDs that can be further resolved
+by other resolvers.
+
+The Connect features support advanced features for being able to then traverse
+this graph via queries is useful ways.
+
+For example, here is how you might generate the support for queries about
+people and their addresses.
+
+While reading this, note that the join implied from person to address is an artifact
+of the *resolver*, *not* the *database* (and thus is independent of the schema
+of the real database):
+
+```
+(ns sample-query-processing
+  (:require
+    [com.wsscode.pathom.core :as p]
+    [com.wsscode.pathom.connect :as pc]))
+
+;; setup for a given connect subsystem
+(defmulti resolver-fn pc/resolver-dispatch)
+(def indexes (atom {}))
+(def defresolver (pc/resolver-factory resolver-fn indexes))
+
+;; How to go from :person/id to that person's details
+(defresolver `person-resolver
+  ;; The minimum data we must already know in order to resolve the outputs
+  {::pc/input  #{:person/id}
+   ;; A query template for what this resolver outputs
+   ::pc/output [:person/name {:person/address [:address/id]}]}
+  (fn [env {:keys [person/id] :as params}]
+    ;; normally you'd pull the person from the db, and satisfy the listed
+    ;; outputs. For demo, we just always return the same person details.
+    {:person/name "Tom"
+     :person/address {:address/id 1}}))
```

Comment:
I would like to discuss this example, this is a case of a to-one relationship, when we have this situation (and is truly to one, in case the user will have to have just a single address, never more) it's better to flatten the map, so instead just add the `:address/id` on the same entity, this makes for a more flexible result (so you can walk directly to many derivatives from the address without changing any context). So doing a nest for to-one is a well known thing but not ideal in Pathom terms. So my suggestion is that or we flatten this, or make this a to-many.